lib/cursor.js

```diff
@@ -291,13 +290,16 @@ Cursor.prototype.filter = function(filter) {
  * Set the cursor maxScan
  * @method
  * @param {object} maxScan Constrains the query to only scan the specified number of documents when fulfilling the query
+ * @deprecated as of MongoDB 4.0
  * @return {Cursor}
  */
 Cursor.prototype.maxScan = function(maxScan) {
   if (this.s.state === Cursor.CLOSED || this.s.state === Cursor.OPEN || this.isDead()) {
     throw MongoError.create({ message: 'Cursor is closed', driver: true });
   }
 
+  console.warn('Cursor maxScan is deprecated, and will be removed in a later version')
```

Comment:
is it worth adding an explicit `deprecate` utility function, since we are likely to need this in many places?

Reply:
I was hoping to make that part of options management, as either an option on the `arity` functions, or a separate function. That also might be a good `neweng` project.